### ISSUE TITLE
The code about loss part.

### ISSUE THREAD

**Issue by Li-Whasaka** [qcraftai/distill-bev#6]
Hi, I have a few questions about the loss section while studying your code, So I need your help to slove these questions:
(1) Is the code and configuration file you have publicly shared the same as the code used to produce the results presented in your research paper?  

(2)In **_CLASS BEVDetDistill_**, function **_distill_loss_**, _**self.distill_type**_  has many options, such as  **_"all, foreground_background,... "_**, the config file have chosen 'fgd', and it seems to contain all the others loss internally, the loss 'all' is not what his name represents.

(3)In **_CLASS BEVDetDistill_**, function **_fgd_distill_loss_**, [**_t_attention_mask,s_attention_mask,c_t_attention_mask_**] in your code seem to be used for the **_Spatial Attention_** mentioned in paper, **_the t_atten..and s_atten.._**  look like have achieved it  , I can't figure out  what's the role of **_c_t_attention_mask_**, and why **_fg_mask * c_t_aten.._** and the kd_channel_loss. 
and
 **_c_t_attention_mask = torch.mean(torch.abs(teacher_feat), [2, 3], keepdim=True)_**,
 **_t_attention_mask = torch.mean(torch.abs(teacher_feat), [1], keepdim=True)_**      
the params **_[1]_** and **_[2,3]_**,which is actually the channel op.
![image](https://github.com/qcraftai/distill-bev/assets/59135190/d8837723-8fd5-41e2-a1be-d93be17326b2)

(4) have you tried distillation with some heatmap overlaid to a certain location ops?
(5)the bevdepth mAP mentioned in your paper is 36.4, Is this(BEVDepth4D-R50-FP16*) being used?

**Comment by dingwenli** [comment]
Hi, you can find the responses below.

For (1), the code we published contains some experimental components, which are not included in the paper.
For (2), yes, fgd contains the componenets used in the paper and ablation study. 'all' is a placeholder.
For (3), the channel attention (c_t_attention_mask) and its associated loss (kd_channel_loss) are not used to reproduce the results in the paper. As you can find in the training scripts and configs (scripts/teacher_to_bevdepth4d/centerpoint2bevdepth.sh and mvp2bevdepth.sh; configs/lidar2camera_bev_distillation/teacher_to_bevformer/lidarformer_to_bevformer_nus_1x1conv_r50.py and mvpformer_to_bevformer_nus_1x1conv_r50.py), the channel mask is turned off.
For (4), we didn't try this option.
For (5), we trained the baseline using bevdepth4d-r50, not the fp16 version.

**Comment by Li-Whasaka** [comment]
> Hi, you can find the responses below.
> 
> For (1), the code we published contains some experimental components, which are not included in the paper. For (2), yes, fgd contains the componenets used in the paper and ablation study. 'all' is a placeholder. For (3), the channel attention (c_t_attention_mask) and its associated loss (kd_channel_loss) are not used to reproduce the results in the paper. As you can find in the training scripts and configs (scripts/teacher_to_bevdepth4d/centerpoint2bevdepth.sh and mvp2bevdepth.sh; configs/lidar2camera_bev_distillation/teacher_to_bevformer/lidarformer_to_bevformer_nus_1x1conv_r50.py and mvpformer_to_bevformer_nus_1x1conv_r50.py), the channel mask is turned off. For (4), we didn't try this option. For (5), we trained the baseline using bevdepth4d-r50, not the fp16 version.

HELPFUL,THANKS A LOT
